### PR TITLE
Create Blog “greenlake-api-how-to-generate-aceess-token-in-python”

### PR DESCRIPTION
**Pull request by buzzsubash**
Automatically generated by Netlify CMS

### CODE REVIEW

**Review by choukden — CHANGES_REQUESTED** [review]
Hi Subash

Thank you very much for your contribution to the HPE Developer Community.

Please find here some comments and suggestions for you to consider and incorporate in the blog post. Once you have incorporated the suggestions (see tips below), I will engage with the HPE GreenLake Doc Writer team for the Editorial/Grammatical review before publishing the post to the HPE Developer Community portal.

I will also need to get your company's e-mail address in order for me to communicate with you via e-mail. 
My email address is denis.choukroun@hpe.com

Some tips to review the suggestions and acknowledge them in GitHub:

- Navigate to **_File Changed_** area to visualize the comments from start to the end, or navigate to **_Conversation_** area to visualize the comments per reviewer.

- You can mark the suggestion you have incorporated in your platform page entry in the CMS Workflow Editor by clicking on button **Resolve Conversation.**

Note we recommend you continue doing edits of your platform entry in the CMS Workflow editor while the entry is in the "**_In Review_**" column. Moving it back to the "**_Drafts_**" column would delete the Pull Request in GitHub.

Thanks!

Denis

**Comment by buzzsubash** [comment]
Hi, I have update the article and made changes as advised.

File: content/blog/greenlake-api-how-to-generate-aceess-token-in-python.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+  - hpe-greenlake
+  - API
+---
+Accessing and organizing audit logs is crucial for keeping your system secure, meeting compliance requirements, and running operations smoothly. HPE GreenLake’s audit logs are essential for tracking activities and spotting any unusual behavior that might indicate a security threat.
+
+In this guide, we'll show you how to set up a Python script to automatically fetch and process HPE GreenLake audit logs. This will save you time and make it easier to keep an eye on your system.
+
+If you’re new to HPE GreenLake’s APIs, take a look at this [blog post](https://developer.hpe.com/blog/get-started-with-the-foundational-apis-for-the-hpe-greenlake-edge-to-cloud-platform-%E2%80%93-part-3-tracking-activities-and-monitoring-health/) to get started with the HPE GreenLake platform API Audit Logs.
+
+## Use cases of fetching audit logs
```
Comment (joe-boyle24) [inline content/blog/greenlake-api-how-to-generate-aceess-token-in-python.md:19]:
```suggestion
## Use cases for fetching audit logs
```

File: content/blog/greenlake-api-how-to-generate-aceess-token-in-python.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+* **Security Monitoring**: Keeping an eye on who accessed the system and what they did helps in spotting unusual activities that might be security threats.
+* **Compliance Auditing**: Many industries have rules about keeping records of system activities. Audit logs are perfect for this and can be shown as proof during audits.
+* **Operational Analysis**: By looking at the audit logs, companies can see patterns in how their systems are used, helping them to make better decisions about resources and planning.
+
+## Detailed steps to set up the automation
+
+Here’s a step-by-step guide to getting your audit logs automatically using Python:
+
+### Step 1: Set up oauth2 authentication
```
Comment (joe-boyle24) [inline content/blog/greenlake-api-how-to-generate-aceess-token-in-python.md:31]:
```suggestion
### Step 1: Set up OAuth2 authentication
```

File: content/blog/greenlake-api-how-to-generate-aceess-token-in-python.md [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+
+HPE GreenLake uses OAuth2 for security, which means you first need to get an access token.
+
+1. **Store Your Credentials**: Keep your client ID and secret safe but accessible to your script. A good practice is to store them in a text file. (In this case, I did save the client_Id in first line and client_secret in second line)
```
Comment (joe-boyle24) [inline content/blog/greenlake-api-how-to-generate-aceess-token-in-python.md:35]:
```suggestion
1. **Store Your Credentials**: Keep your client ID and secret safe but accessible to your script. A good practice is to store them in a text file. In this case, I did save the client_Id in first line and client_secret in second line.
```

File: content/blog/greenlake-api-how-to-generate-aceess-token-in-python.md [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+        client_id, client_secret = [line.strip() for line in file.readlines()]
+    client = BackendApplicationClient(client_id=client_id)
+    oauth = OAuth2Session(client=client)
+    token = oauth.fetch_token(token_url='https://sso.common.cloud.hpe.com/as/token.oauth2', auth=(client_id, client_secret))
+    return token['access_token']
+```
+
+### Step 2: Fetch the audit logs
+
+Now that you have a token, you can ask HPE GreenLake for the logs.
```
Comment (joe-boyle24) [inline content/blog/greenlake-api-how-to-generate-aceess-token-in-python.md:50]:
```suggestion
Now that you have a token, you can fetch the logs from HPE GreenLake.
```

File: content/blog/greenlake-api-how-to-generate-aceess-token-in-python.md [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+```python
+def fetch_audit_logs(token):
+    headers = {'Authorization': f'Bearer {token}'}
+    response = requests.get('https://global.api.greenlake.hpe.com/audit-log/v1beta1/logs', headers=headers)
+    return response.json() if response.status_code == 200 else None
+```
+
+### Step 3: Process and save the data
+
+After getting the logs, you'll want to turn them into a format that’s easy to read and use.
```
Comment (joe-boyle24) [inline content/blog/greenlake-api-how-to-generate-aceess-token-in-python.md:63]:
```suggestion
After getting the logs, you'll want to convert them into a format that’s easy to read and use.
```

File: content/blog/greenlake-api-how-to-generate-aceess-token-in-python.md [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+
+* **Process the JSON**: Convert the JSON data into a pandas DataFrame. This makes it easier to work with in Python.
+* **Save to Excel**: Export this DataFrame to an Excel file. This file will be your audit log record that you can open, share, and analyze without needing specialised software.
+
+## How does it look like in excel?
```
Comment (joe-boyle24) [inline content/blog/greenlake-api-how-to-generate-aceess-token-in-python.md:68]:
```suggestion
## What does it look like in excel?
```

File: content/blog/greenlake-api-how-to-generate-aceess-token-in-python.md [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+
+![](/img/excel-output.png)
+
+I have used excel in this script just for testing and can be used with any of the databases including sqlite, postgres and oracle, and yes, python can do it, however, it varies on different use case, especially when it comes to real-time data monitoring or handling very large volumes of logs.
```
Comment (joe-boyle24) [inline content/blog/greenlake-api-how-to-generate-aceess-token-in-python.md:72]:
```suggestion
I have used excel in this script just for testing and can be used with any of the databases including sqlite, postgres and oracle. Python can also do it, but the results vary depending on the use case, especially when it comes to real-time data monitoring or handling very large volumes of logs.
```

**Review by joe-boyle24 — CHANGES_REQUESTED** [review]

**Comment by buzzsubash** [comment]
@joe-boyle24 Thank you, updated all suggestions

**Pull request merged**